cassovary-core/src/test/scala/com/twitter/cassovary/algorithms/linkanalysis/PageRankSpec.scala

```diff
@@ -91,7 +91,17 @@ class PageRankSpec extends WordSpec with Matchers {
       val pr = new PageRank(graphComplete, params)
 
       val pagerank = pr.run().pageRank
-      graphComplete.foreach { n => pagerank(n.id) shouldEqual 0.1 }
+      graphComplete.foreach { n => pagerank(n.id) should be (0.1 +- 1E-8) }
+    }
+
+    "Handle dangling nodes correctly when graph is stored only out" in {
+      val target = Array(0.0, 0.11622, 0.21500, 0.16763, 0.16763, 0.07625, 0.14106, 0.11622)
+      testGraphHelper(TestGraphs.dangling_g7, target)
+    }
+
+    "Handle dangling nodes correctly when graph is stored only in" in {
+      val target = Array(0.0, 0.12085, 0.15194, 0.05628, 0.05628, 0.28697, 0.27141, 0.05628)
+      testGraphHelper(TestGraphs.dangling_g7_in, target)
     }
```

Comment:
can you have a very small say 3-node graph A->C, B->C (i.e., node B is zeroOutgoing) to verify the exact pagerank values in the test?